qiskit/ignis/calibrations/schedules.py

```diff
@@ -0,0 +1,187 @@
+                   meas_map=None):
+    """
+    Generates schedules for a rabi experiment using a Gaussian pulse
+
+    Args:
+        amp_list (list of floats): List of amplitudes for the Gaussian
+        pulse [-1,1]
+        qubits (list of integers): indices of the qubits to perform a rabi
+        pulse_width: width of gaussian (in dt units)
+        pulse_sigma: sigma of gaussian
+        width_sigma_ratio: set sigma to a certain ratio of the width (use if
+        pulse_sigma is None)
+        system: PulseChannelSpec object
+        cmd_def: CmdDef object to use
+        meas_map: meas_map to use
+
+    Returns:
+       A list of QuantumSchedules
+       xdata: a list of amps
+    """
+
+    xdata = amp_list
+
+    # copy the command def
+    cmd_def = copy.deepcopy(cmd_def)
+
+    if pulse_sigma is None:
+        pulse_sigma = pulse_width / width_sigma_ratio
+
+    # Construct the circuits
+    qr = qiskit.QuantumRegister(max(qubits) + 1)
+    cr = qiskit.ClassicalRegister(len(qubits))
+
+    circuits = []
+
+    for circ_index, g_amp in enumerate(amp_list):
+
+        circ = qiskit.QuantumCircuit(qr, cr)
+        circ.name = 'rabicircuit_%d_0' % circ_index
+
+        rabi_pulse = pulse_lib.gaussian(duration=pulse_width,
+                                        amp=g_amp,
+                                        sigma=pulse_sigma,
+                                        name='rabi_pulse_%d' % circ_index)
+
+        rabi_gate = Gate(name='rabi_%d' % circ_index, num_qubits=1, params=[])
+
+        for _, qubit in enumerate(qubits):
+
+            # add commands to schedule
+            schedule = pulse.Schedule(name='rabi_pulse_%f_%d' % (g_amp,
+                                                                 qubit))
+
+            schedule += rabi_pulse(system.qubits[qubit].drive)
+
+            # append this schedule to the cmd_def
+            cmd_def.add('rabi_%d' % circ_index, qubits=[qubit],
+                        schedule=schedule)
+
+            circ.append(rabi_gate, [qr[qubit]])
+
+        for qind, qubit in enumerate(qubits):
+            circ.measure(qr[qubit], cr[qind])
+
+        circuits.append(circ)
+
+        # schedule
+        schedule_config = ScheduleConfig(cmd_def, meas_map)
+        rabi_sched = [schedule_circuit(qcirc,
+                                       schedule_config)
+                      for qcirc in circuits]
+
+    return rabi_sched, xdata
+
+
+def drag_schedules(beta_list, qubits, pulse_amp, pulse_width,
+                   pulse_sigma=None,
+                   width_sigma_ratio=4, system=None, cmd_def=None,
+                   meas_map=None):
+    """
+    Generates schedules for a drag experiment doing a pulse then
+    the - pulse
+
+    Args:
+        beta_list (list of floats): List of relative amplitudes
+        for the derivative pulse
+        qubits (list of integers): indices of the qubits to perform a rabi
+        pulse_amp: amp of the gaussian (list of length qubits)
```

Comment:
```suggestion
        pulse_amp: Previously fitted amp of the gaussian (list of length qubits)
```

This is another instance of something that could be per-qubit.